cli_tools/common/domain/interfaces.go

```diff
@@ -66,9 +65,20 @@ type TarGcsExtractorInterface interface {
 	ExtractTarToGcs(tarGcsPath string, destinationGcsPath string) error
 }
 
-// StorageObjectDeleterInterface represents an object that is responsible for deleting GCS objects
-type StorageObjectDeleterInterface interface {
-	DeleteObject(bucket string, objectPath string) error
+// StorageObjectCreatorInterface represents GCS object creator
+type StorageObjectCreatorInterface interface {
+	GetObject(bucket string, objectPath string) StorageObjectInterface
+}
```

Comment:
What is the need for this interface? Can it be integrated into StorageClient (so that StorageClient.GetObject does the reference creation)?

Also, this only creates a reference to a GCS object, doesn't actually create anything in GCS so the naming doesn't seem appropriate.

Reply:
I think this is used so that we can mock the object in storage client tests.

Reply:
I think the interface is needed to that we can mock this in storage client tests.

Reply:
StorageObjectInterface is better

Reply:
> What is the need for this interface? 

It's typically an anti-pattern to expose your implementation details (unless there's an overwhelmeing reason for it). Some good reading: https://golang.org/doc/effective_go.html#generality